rollups.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/signalfx/golib/sfxclient"
+)
+
+// Rollups rolls up data and sends it periodically to signalfx.
+type Rollups struct {
+	sfxSink                            sfxclient.Sink
+	scheduler                          *sfxclient.Scheduler
+	rollupsRequestFinishedResponseTime map[string]*sfxclient.RollingBucket
+	rollupsRequestFinishedStatusCode   map[string]*sfxclient.CumulativeBucket
+}
+
+func NewRollups(sfxSink sfxclient.Sink) *Rollups {
+	scheduler := sfxclient.NewScheduler()
+	scheduler.Sink = sfxSink
+	scheduler.SendZeroTime = true
+	scheduler.ErrorHandler = func(err error) error {
+		lg.ErrorD("rollup-error", map[string]interface{}{"error": err.Error()})
+		return nil
+	}
+	return &Rollups{
+		sfxSink:                            sfxSink,
+		scheduler:                          scheduler,
+		rollupsRequestFinishedResponseTime: map[string]*sfxclient.RollingBucket{},
+		rollupsRequestFinishedStatusCode:   map[string]*sfxclient.CumulativeBucket{},
+	}
+}
+
+// Run this in a goroutine.
+func (r *Rollups) Run(ctx context.Context) error {
+	return r.scheduler.Schedule(ctx)
+}
+
+func (r *Rollups) Process(fields map[string]interface{}) error {
+	if fields["env"] != "" && fields["container_app"] != "" && fields["title"] == "request-finished" && fields["via"] == "kayvee-middleware" {
+
+		// rollup response times
+		{
+			dimensions := map[string]string{
+				"env":           fields["env"].(string),
+				"container_app": fields["container_app"].(string),
+				"canary":        fmt.Sprintf("%t", fields["canary"].(bool)),
+			}
+			dimensionsKey := join(dimensions, "-")
+			bucket, ok := r.rollupsRequestFinishedResponseTime[dimensionsKey]
+			if !ok {
+				lg.InfoD("rollup-create", map[string]interface{}{"key": dimensionsKey})
```

Comment:
nit: include metricName "request-finished-response-time" here (and same idea in below info log) so it's easier to tell which rollup was created